Modules/System/Azure Queues/src/AzureStorageQueuesImpl.Codeunit.al

```diff
@@ -0,0 +1,293 @@
+    var
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        HttpRequest: HttpRequestMessage;
+        HttpResponse: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        HttpClient.Delete(GetAzureQueueUpdateMessageEndpoint(StorageAccountName, Queue, MessageID, PopReceipt), HttpResponse);
+        exit(HttpResponse.IsSuccessStatusCode);
+    end;
+
+    procedure ClearMessagesFromQueue(StorageAccountName: Text; Queue: Text[20]): Boolean
+    var
+        HttpClient: HttpClient;
+        HttpHeaders: HttpHeaders;
+        HttpContent: HttpContent;
+        HttpRequest: HttpRequestMessage;
+        HttpResponse: HttpResponseMessage;
+        ResponseText: Text;
+    begin
+        HttpClient.Delete(GetAzureQueueMessagesEndpoint(StorageAccountName, Queue), HttpResponse);
+        exit(HttpResponse.IsSuccessStatusCode);
+    end;
+
+    local procedure GetQueueEndpoint(StorageAccountName: Text; Queue: Text): Text
+    var
+        AzureQueueSetup: Record "Azure Queue Setup";
+        EndPoint: Text;
+    begin
+        AzureQueueSetup.Get(StorageAccountName);
+        Endpoint := GetAzureQueueBaseEndpoint(StorageAccountName);
+        exit(EndPoint + '/' + Queue + '/' + AzureQueueSetup."SAS Key");
+    end;
+
+    local procedure GetAzureQueueServiceEndpoint(StorageAccountName: Text): Text
+    var
+        AzureQueueSetup: Record "Azure Queue Setup";
+        EndPoint: Text;
+    begin
+        AzureQueueSetup.Get(StorageAccountName);
+        Endpoint := GetAzureQueueBaseEndpoint(StorageAccountName);
+        exit(EndPoint + AzureQueueSetup."SAS Key");
+    end;
+
+    local procedure GetAzureQueueUpdateMessageEndpoint(StorageAccountName: Text; Queue: Text; MessageID: Text; Popreceipt: Text): Text
+    var
+        AzureQueueSetup: Record "Azure Queue Setup";
+        EndPoint: Text;
+    begin
+        AzureQueueSetup.Get(StorageAccountName);
+        Endpoint := GetAzureQueueBaseEndpoint(StorageAccountName);
+        exit(EndPoint + '/' + Queue + '/' + 'messages' + '/' + MessageID + AzureQueueSetup."SAS Key" + '&' + 'popreceipt=' + Popreceipt);
+
+    end;
+
+    local procedure GetAzureQueueMessagesEndpoint(StorageAccountName: Text; Queue: Text): Text
+    var
+        AzureQueueSetup: Record "Azure Queue Setup";
+        EndPoint: Text;
+    begin
+        AzureQueueSetup.Get(StorageAccountName);
+        Endpoint := GetAzureQueueBaseEndpoint(StorageAccountName);
+        exit(EndPoint + '/' + Queue + '/' + 'messages' + AzureQueueSetup."SAS Key");
+    end;
+
+    local procedure GetAzureQueuePeekMessagesEndpoint(StorageAccountName: Text; Queue: Text): Text
+    var
+        AzureQueueSetup: Record "Azure Queue Setup";
+        EndPoint: Text;
+    begin
+        AzureQueueSetup.Get(StorageAccountName);
+        Endpoint := GetAzureQueueBaseEndpoint(StorageAccountName);
```

Comment:
Maybe this methods could be improved by using the URI Codeunit to create URI instead of simple concatenating base path and subpaths.

Reply:
@pri-kise That would make the code more robust for sure